eng/common/native/init-compiler.sh

```diff
@@ -71,7 +71,7 @@ if [[ -z "$CLR_CC" ]]; then
     # Set default versions
     if [[ -z "$majorVersion" ]]; then
         # note: gcc (all versions) and clang versions higher than 6 do not have minor version in file name, if it is zero.
-        if [[ "$compiler" == "clang" ]]; then versions=( 15 14 13 12 11 10 9 8 7 6.0 5.0 4.0 3.9 3.8 3.7 3.6 3.5 )
+        if [[ "$compiler" == "clang" ]]; then versions=( 16 15 14 13 12 11 10 9 8 7 6.0 5.0 4.0 3.9 3.8 3.7 3.6 3.5 )
```

Comment:
This needs to be also in release/7.0 branch of arcade repo: https://github.com/dotnet/arcade/blob/release/7.0/eng/common/native/init-compiler.sh

Reply:
Done: https://github.com/dotnet/arcade/pull/13062